processing/src/main/java/org/apache/druid/segment/nested/ScalarDoubleColumn.java

```diff
@@ -91,6 +92,9 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
       @Override
       public boolean isNull()
       {
+        if (NullHandling.replaceWithDefault()) {
+          return false;
+        }
```

Comment:
Instead of doing the `NullHandling` check here, can you make it a field and have the Supplier pass it in?